### PR TITLE
DISCO-1399: Deprecating pre-type workflow

### PR DESCRIPTION
**Pull request by Dillon-Dumesnil**

### CODE REVIEW

File: course_discovery/apps/api/v1/views/courses.py [hunk 1]
```diff
@@ -296,19 +292,18 @@ def update_course(self, data, partial=False):  # pylint: disable=too-many-statem
         serializer.is_valid(raise_exception=True)
 
         # First, update course entitlements
-        if data.get('type') or data.get('prices'):
```
Comment (mikix) [inline course_discovery/apps/api/v1/views/courses.py]: This line is still useful - you might not PATCH a type or price (I know FAP always does, but as an endpoint...)

**Review by mikix — APPROVED** [review]

**Comment by mikix** [comment]
While digging around, I found a spot in could_be_marketable and is_marketable that could use cleaning.

File: course_discovery/apps/api/v1/views/courses.py [hunk 2]
```diff
@@ -275,16 +275,12 @@ def update_entitlement(self, course, entitlement_type, price, partial=False):
                 draft=True,
             ), True)
 
-    # DISCO-1399: You might be able to removew the too-many-statements disable since it should get rid
-    # of a number of if statements since we don't need to check for `if data.get('type')`
     @writable_request_wrapper
-    def update_course(self, data, partial=False):  # pylint: disable=too-many-statements
```
Comment (mikix) [inline course_discovery/apps/api/v1/views/courses.py:281]: Hah were those two lines below enough?

File: course_discovery/apps/course_metadata/management/commands/load_program_fixture.py [hunk 3]
```diff
@@ -13,13 +13,12 @@
...
-    CourseRun, Curriculum, CurriculumCourseMembership, CurriculumProgramMembership, Program, ProgramType, Seat,
-    SeatType
+    CourseRun, Curriculum, CurriculumCourseMembership, CurriculumProgramMembership, Program, ProgramType, SeatType
 )
 from course_discovery.apps.course_metadata.signals import (
-    add_masters_track_on_course, check_curriculum_for_cycles, check_curriculum_program_membership_for_cycles,
+    check_curriculum_for_cycles, check_curriculum_program_membership_for_cycles,
     ensure_external_key_uniquness__course_run, ensure_external_key_uniquness__curriculum,
-    ensure_external_key_uniquness__curriculum_course_membership, publish_masters_track
+    ensure_external_key_uniquness__curriculum_course_membership
```
Comment (mikix) [inline course_discovery/apps/course_metadata/management/commands/load_program_fixture.py]: Ugh that typo.

File: course_discovery/apps/course_metadata/signals.py [hunk 4]
```diff
@@ -40,77 +37,6 @@ def is_program_masters(program):
...
-
-        if not is_program_masters(program):
-            logger.debug('This is a course related only to non-masters program. No need to create Masters seat')
-            return
-
-        us_currency = Currency.objects.get(code='USD')
-
-        for course_run in instance.course_runs:
-            if course_run.type:
-                # We are in the middle of a transition to CourseRunType. If we have a type, we handle this elsewhere.
```
Comment (mikix) [inline course_discovery/apps/course_metadata/signals.py:61]: I really should have marked this with DISCO-1399. Sorry I wasn't as thorough about that around the place.

File: course_discovery/apps/course_metadata/utils.py [hunk 5]
```diff
@@ -552,152 +548,6 @@ def push_tracks_to_lms_for_course_run(course_run):
                            response.content.decode('utf-8'))
 
 
-@transaction.atomic
-def publish_to_course_metadata(partner, course_run, create_official=True, fail_on_url_slug=True):
```
Comment (mikix) [inline course_discovery/apps/course_metadata/utils.py:556]: Ahhhhhh

File: course_discovery/apps/publisher/api/v1/views.py [hunk 6]
```diff
@@ -82,6 +82,5 @@ def publish_to_ecommerce(self, course_run):
             logger.exception('Failed to publish course run [%d] to E-Commerce!', course_run.pk)
             return 'FAILED: ' + str(ex)
 
-    def publish_to_discovery(self, partner, course_run):
-        publish_to_course_metadata(partner, course_run)
+    def publish_to_discovery(self):
         return self.PUBLICATION_SUCCESS_STATUS
```
Comment (mikix) [inline course_discovery/apps/publisher/api/v1/views.py:86]: Savage. "Got it, boss!"

**Review by mikix — APPROVED** [review]
Love it.

File: course_discovery/apps/course_metadata/management/commands/load_program_fixture.py (hunk 3 above)
Comment (Dillon-Dumesnil) [reply course_discovery/apps/course_metadata/management/commands/load_program_fixture.py]: Fixed it for all three of these functions.

File: course_discovery/apps/api/v1/views/courses.py (hunk 2 above)
Comment (Dillon-Dumesnil) [reply course_discovery/apps/api/v1/views/courses.py:281]: Luckily!

File: course_discovery/apps/publisher/api/v1/views.py (hunk 6 above)
Comment (Dillon-Dumesnil) [reply course_discovery/apps/publisher/api/v1/views.py:86]: Figured I'd just leave it for the person actually deleting that file

**Pull request merged**